### PR TITLE
libsel4utils: fix null pointer reference

### PR DESCRIPTION
**Pull request by anchao**
libsel4utils: fix null pointer reference

add sanity check after alloc irq_server_t

### CODE REVIEW

**Review by axel-h — APPROVED** [review]
Thanks, looks good to me

File: libsel4utils/src/irq_server/irq_server.c [hunk 1]
```diff
@@ -349,6 +354,7 @@ irq_server_t *irq_server_new(vspace_t *vspace, vka_t *vka, seL4_Word priority,
     error = sel4platsupport_new_irq_ops(&(new->irq_ops), vka, simple, irq_config, malloc_ops);
     if (error) {
         ZF_LOGE("Failed to initialise supporting backend for IRQ server");
+        ps_free(malloc_ops, sizeof(irq_server_t), new);
```
Comment (Indanz) [inline libsel4utils/src/irq_server/irq_server.c:357]: Shouldn't you free `new->reply` first?

**Review by Indanz — APPROVED** [review]
Otherwise seems fine.

File: libsel4utils/src/irq_server/irq_server.c (hunk 1 above)
Comment (anchao) [reply libsel4utils/src/irq_server/irq_server.c:359]: Done, thank you for review

**Pull request merged**